analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -25,6 +25,17 @@ namespace SonarAnalyzer.AnalysisContext;
 
 public class SonarAnalysisContext
 {
+    private readonly HashSet<string> rulesDisabledForRazor = new()
+        {
+            "S103",
+            "S104",
+            "S109",
```

Comment:
I am surprised we don't have many S109 as expected issues in the ITs.

Reply:
The hoard of S109 issues come from lines like `__builder3.AddContent(13, "Not found");` where the literal 13 is mapped back to the .razor/.cshtml file even if it shouldn't due to [this](https://github.com/dotnet/roslyn/issues/69248)  bug on roslyn side.

Reply:
Since, the .razor and .cshtml files are very minimal in the ITs, there are no  such addcontent calls and no issues.
I would not complicate the ITs as the functionality is already tested in the UTs and we will also have peach to validate them.